src/ssh.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+    SPDX-FileCopyrightText: Lieven Hey <lieven.hey@kdab.com>
+    SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+
+    SPDX-License-Identifier: GPL-2.0-or-later
+*/
+
+#include "ssh.h"
+
+#include <QProcess>
+
+#include <KConfigGroup>
+#include <KSharedConfig>
+
+#include "settings.h"
+#include "util.h"
+
+QStringList assembleSSHArguments(const QString& deviceName, const QStringList& command)
+{
+    KConfigGroup device = KSharedConfig::openConfig()->group("devices").group(deviceName);
+
+    QStringList arguments = {
+        QStringLiteral("%1@%2").arg(device.readEntry("username", QString()), device.readEntry("hostname", QString()))};
+    const auto sshOptions = device.readEntry("sshoptions", QString());
+    if (!sshOptions.isEmpty()) {
+        arguments.append(sshOptions.split(QLatin1Char(' ')));
+    }
+
+    arguments.append(command);
+
+    return arguments;
+}
+
+QProcessEnvironment sshEnvironment()
+{
+    auto env = Util::appImageEnvironment();
+    Settings* settings = Settings::instance();
+    env.insert(QLatin1String("SSH_ASKPASS"), settings->sshaskpassPath());
```

Comment:
QStringLiteral

and what about people like me who actually set this env var already but haven't configured it in hotspot? this is all backwards

Reply:
I added a check to only set it, if `settings->sshAskPassPath()` has a value.